packages/confect/src/server/ConfectApiServer.ts

```diff
@@ -402,3 +418,62 @@ const confectActionFunction = <
       Effect.runPromise,
     ),
 });
+
+export type ServerApi<Spec extends ConfectApiSpec.ConfectApiSpec.AnyWithProps> =
+  Types.Simplify<Helper<ConfectApiSpec.ConfectApiSpec.Groups<Spec>>>;
+
+type Helper<Groups extends ConfectApiGroup.ConfectApiGroup.AnyWithProps> = {
+  [GroupName in ConfectApiGroup.ConfectApiGroup.Name<Groups>]: ConfectApiGroup.ConfectApiGroup.WithName<
+    Groups,
+    GroupName
+  > extends infer Group extends ConfectApiGroup.ConfectApiGroup.AnyWithProps
+    ? ConfectApiGroup.ConfectApiGroup.Groups<Group> extends infer SubGroups extends
+        ConfectApiGroup.ConfectApiGroup.AnyWithProps
+      ? Types.Simplify<
+          Helper<SubGroups> & {
+            [FunctionName in ConfectApiFunction.ConfectApiFunction.Name<
+              ConfectApiGroup.ConfectApiGroup.Functions<Group>
+            >]: ConfectApiFunction.ConfectApiFunction.WithName<
+              ConfectApiGroup.ConfectApiGroup.Functions<Group>,
+              FunctionName
+            > extends infer Function extends
+              ConfectApiFunction.ConfectApiFunction.AnyWithProps
+              ? RegisteredFunctionFromConfectFunction<Function>
+              : never;
+          }
+        >
+      : {
+          [FunctionName in ConfectApiFunction.ConfectApiFunction.Name<
+            ConfectApiGroup.ConfectApiGroup.Functions<Group>
+          >]: ConfectApiFunction.ConfectApiFunction.WithName<
+            ConfectApiGroup.ConfectApiGroup.Functions<Group>,
+            FunctionName
+          > extends infer Function extends
+            ConfectApiFunction.ConfectApiFunction.AnyWithProps
+            ? RegisteredFunctionFromConfectFunction<Function>
+            : never;
+        }
+    : never;
+};
+
+type RegisteredFunctionFromConfectFunction<
+  Function extends ConfectApiFunction.ConfectApiFunction.AnyWithProps,
+> = Function["functionType"] extends "Query"
+  ? RegisteredQuery<
+      Lowercase<Function["functionVisibility"]>,
+      Function["args"]["Encoded"],
+      Promise<Function["returns"]["Encoded"]>
+    >
+  : Function["functionType"] extends "Mutation"
+    ? RegisteredMutation<
+        Lowercase<Function["functionVisibility"]>,
+        Function["args"]["Encoded"],
+        Promise<Function["returns"]["Encoded"]>
+      >
+    : Function["functionType"] extends "Action"
+      ? RegisteredAction<
+          Lowercase<Function["functionVisibility"]>,
+          Function["args"]["Encoded"],
+          Promise<Function["returns"]["Encoded"]>
+        >
+      : never;
```

Comment:
Let's move this to the `ConfectApiFunction` namespace in the `ConfectApiFunction` module, and just call it `RegisteredFunction` (think of it as a "getter" type). You could also use the other "getter" types in that module in this type body, e.g. `Args<Function>["Encoded"]` over `Function["args"]["Encoded"]`.